### DISCUSSION TITLE
GitHub Actions

### DISCUSSION THREAD

**Discussion by vybhav72954** [HarshCasper/Rotten-Scripts#672] [Ideas]
Hey, @HarshCasper A few questions, please.

I can either tun the whole linter on the whole codebase (basically meaning that all the files existing will be dealt with), but if I will ensure this functionality, it will be invoked every time, hence not only will make the action super slow but might result in multiple errors (assuming that a lot of files might be poorly formatted).

Or I can run the linter on the new files (new PRs only), remotely run my Linter on the whole repo, make issues for correcting those READMEs, I will deal with as many as I can.

Tell me how you feel about this though.

Secondly, I noticed that you have been pointing out `docstrings` and `shebang` in the code reviews, I will make my PyLint action more strict as it is not strict at all right now.

**Comment by vybhav72954** [comment]
@HarshCasper Do we need a report or a log file as well? Though I doubt as it will cram up the Repo.

**Comment by HarshCasper** [comment]
You can turn the Linter on the new PRs right now. Some of the existing Code is poorly formatted and I want to focus on cleaning them first, before we turn our attention towards having new enhancements. Regarding a strict PyLint, I have already commented on the Issue. Feel free to use whatever you may like. 

Log File is not necessary. It is the least thing that I would like to deal with everyday 😆

**Comment by vybhav72954** [comment]
Dear @HarshCasper 

I was out of the station and was inactive for 2 weeks. I was just going to finalize the Action.

i am facing one problem, i.e. [this](https://pasteboard.co/JJILYXC.png)

In this case, the action took about 5 min to complete, while I pushed a single file

Be advised that with multiple files in a commit or a PR where the action needs to find the `md` file first, this will take  longer, the problem is simple.


My action is doing 4 things as of now

- Styling of Markdown
- Spell Checking
- Link Checking
- Styling of YAML

Though we can use GitHub pipelining to increase the speed, hence I suggest to break thin into 3 actions, and then this will take approx 2 min to complete all the task, is that fine by you?

**Reply by HarshCasper** [reply]
Looks good to me!

**Comment by vybhav72954** [comment]
@HarshCasper The PR is open

**Comment by vybhav72954** [comment]
Problem Observed:

- Spell Check is working on every single file (unintentional, bug)

- Markdown Lint is working properly, but we need to make some changes there, it is too strict, and/or we need to make a styling guide so that people can make good README.

- Markdown Link is being weird:
 Reason - 

    1.   The README file I uploaded in #720 is actually horrible, my bad (so basically the action actually worked)
    1.   But there is more to it, I made changes in Python->README and the action hates that. simple reason.
            > The README is formatted like this ```Name of a Repo, Click here[address of repo]```, which is actually fine, but the format of the link is incorrect, say, for example, a repo `xyz` in python.

- Current link format (/Python/xyz)
- Correct link format (./xyz)

The action reports this as an error every time.

Secondly, LinkedIn is directing it to 999 code.
lastly, sometimes GitHub links are redirecting to 429 code [too many requests].

Proposed Action plan:

1. Deal with the Spell  Check ASAP, (working on it)
1. New Styling guide (#714)
1. For markdown link (Waiting for @HarshCasper's guidance)
    - Either ignore every LinkedIn link OR ignores status code 999, [I prefer ignoring LinkedIn, although the 2nd way is quicker to implement]
    - Status Code 429 should be interpreted as 200 only, [can implement that].
    - Ignore the `Python->README` file.

**Reply by HarshCasper** [reply]
Sure. That would be great.

**Reply by vybhav72954** [reply]
> Ignore the Python->README file.

Going to do this, found multiple problems cause of this.

**Comment by vybhav72954** [comment]
I meant what should i do? Cause there are 3 tracks for the markdown link action which I can implement, kindly clear it asap.

**Reply by HarshCasper** [reply]
For now, please work on the Styling Guide. That is definitely on the priority now.

**Reply by vybhav72954** [reply]
So should I leave action as it is?

**Reply by HarshCasper** [reply]
Yeah.

**Reply by vybhav72954** [reply]
I already have a good idea of what i want with the Styling Guide, just a final question, can I also make a few changes in Contribution.md
